agent/agent.go

```diff
@@ -273,8 +312,15 @@ func (r *Agent) setupSink(ctx context.Context, sr recipe.PluginRecipe, stream *s
 			retryNotification,
 		)
 
-		success := err == nil
-		r.monitor.RecordPlugin(recipe.Name, sr.Name, "sink", success)
+		pluginInfo.Success = err == nil
+		for _, val := range stream.subscribers {
+			pluginInfo.BatchSize += val.batchSize
+		}
+
+		for _, monitor := range r.monitor {
+			monitor.RecordPlugin(ctx, pluginInfo)
```

Comment:
This won't work. We would end up recording the sink duration as a continuously increasing value since we are calculating the duration against when this function was called first and the number of retries also wouldn't be correct - we would track 1, 2, 3 retries bringing the total to 6 when the actual number of retries was 3. Instead we should look to implement an `OtelSinkMiddleware` that can emit metrics on each call of `Process`.

Reply:
updated, but still got doubt on batch_size and retry count. Currently adding 1 on every RecordPlugin call